vignettes/design-principles.Rmd

```diff
@@ -0,0 +1,42 @@
+---
+title: "Design Principles for linelist"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Design Principles for linelist}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+This vignette outlines the design decisions that have been taken during the development of the `{linest}` R package, and provides some of the reasoning, and possible pros and cons of each decision.
+
+This document is primarily intended to be read by those interested in understanding the code within the package and for potential package contributors.
+
+## Scope
+
+linelist provides a lightweight layer to add tags to data.frame columns. This allows:
```

Comment:
Backticks for "data.frame"? Possibly also `<>` class notation?

Reply:
Fixed in [`ee556b4` (#112)](https://github.com/epiverse-trace/linelist/pull/112/commits/ee556b4a74777236141baf3fbda35388a2f5a6d1).